app/components/message-input/component.js

```diff
@@ -0,0 +1,68 @@
+import Component from '@ember/component';
+import { bindKeyboardShortcuts, unbindKeyboardShortcuts } from 'ember-keyboard-shortcuts';
+import { isPresent } from '@ember/utils';
+import { scheduleOnce } from '@ember/runloop';
+
+export default Component.extend({
+
+  tagName: 'section',
+  elementId: 'new-message',
+
+  message: '',
+  usernames: null,
+
+  keyboardShortcuts: Object.freeze({
+    'tab': 'completeUsername'
+  }),
+
+  didInsertElement () {
+    this._super(...arguments);
+    bindKeyboardShortcuts(this);
+  },
+
+  willDestroyElement () {
+    this._super(...arguments);
+    unbindKeyboardShortcuts(this);
+  },
+
+  actions: {
+
+    completeUsername () {
+      let input = this.element.querySelector('#message-field');
+      let cursorPosition = input.selectionStart;
+      let textBeforeCursor = this.message.slice(0, cursorPosition);
+      let textAfterCursor = this.message.slice(cursorPosition);
+      let words = textBeforeCursor.split(' ');
+      let searchWord = words.pop();
```

Comment:
I don't think all of these are being re-assigned afterwards, so most of them should be `const`, no?